pkg/messageformat/encoding.go

```diff
@@ -0,0 +1,241 @@
+package messageformat
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+)
+
+// message stores MF2 message.
+// Specification draft for MF2 syntax:
+// https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md
+type message struct {
```

Comment:
Do NOT use multiple bytes.Buffer to encode MF2. Use single only! It is inefficient.